web/client/components/background/BackgroundSelector.jsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2017, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const React = require('react');
+const assign = require('object-assign');
+const {isEmpty} = require('lodash');
+
+const PreviewButton = require('./PreviewButton');
+const PreviewList = require('./PreviewList');
+const PreviewIcon = require('./PreviewIcon');
+
+const HYBRID = require('./images/mapthumbs/HYBRID.jpg');
+const ROADMAP = require('./images/mapthumbs/ROADMAP.jpg');
+const TERRAIN = require('./images/mapthumbs/TERRAIN.jpg');
+const Aerial = require('./images/mapthumbs/Aerial.jpg');
+const mapnik = require('./images/mapthumbs/mapnik.jpg');
+const mapquestOsm = require('./images/mapthumbs/mapquest-osm.jpg');
+const empty = require('./images/mapthumbs/none.jpg');
+const unknown = require('./images/mapthumbs/dafault.jpg');
+const Night2012 = require('./images/mapthumbs/NASA_NIGHT.jpg');
+const AerialWithLabels = require('./images/mapthumbs/AerialWithLabels.jpg');
+const OpenTopoMap = require('./images/mapthumbs/OpenTopoMap.jpg');
+
+const thumbs = {
+    google: {
+        HYBRID,
+        ROADMAP,
+        TERRAIN
+    },
+    bing: {
+        Aerial,
+        AerialWithLabels
+    },
+    osm: {
+        mapnik
+    },
+    mapquest: {
+        osm: mapquestOsm
+    },
+    ol: {
+        "undefined": empty
+    },
+    nasagibs: {
+        Night2012
+    },
+    OpenTopoMap: {
+        OpenTopoMap
+    },
+    unknown
+};
+
+require('./css/background.css');
+
+const BackgroundSelector = React.createClass({
+    propTypes: {
+        drawerWidth: React.PropTypes.number,
+        start: React.PropTypes.number,
+        bottom: React.PropTypes.number,
+        isMobile: React.PropTypes.bool,
+        enabled: React.PropTypes.bool,
+        drawerEnabled: React.PropTypes.bool,
+        layers: React.PropTypes.array,
+        currentLayer: React.PropTypes.object,
+        tempLayer: React.PropTypes.object,
+        size: React.PropTypes.object,
+        desktop: React.PropTypes.object,
+        mobile: React.PropTypes.object,
+        onToggle: React.PropTypes.func,
+        propertiesChangeHandler: React.PropTypes.func,
+        setControlProperty: React.PropTypes.func
+    },
+    getDefaultProps() {
+        return {
+            drawerWidth: 0,
+            start: 0,
+            bottom: 0,
+            isMobile: false,
+            enabled: false,
+            drawerEnabled: false,
+            layers: [],
+            currentLayer: {},
+            tempLayer: {},
+            size: {width: 0, height: 0},
+            desktop: {},
+            mobile: {},
+            onToggle: () => {},
+            propertiesChangeHandler: () => {},
+            setControlProperty: () => {}
+        };
+    },
+    componentWillUnmount() {
+        this.props.setControlProperty('backgroundSelector', 'currentLayer', {});
```

Comment:
use a separate reducer or bind this method to background selector from the plugin. I don't want to make components aware of the setControlProperty method if it is possible.